src/project-conf/index.ts

```diff
@@ -58,32 +58,47 @@ export async function getProjectConfigurationElements(workspaceFolder: Uri) {
 
   const projectConfElements: { [key: string]: ProjectConfElement } = {};
 
-  Object.keys(projectConfJson).forEach((elem) => {
-    projectConfElements[elem] = {
-      build: {
-        compileArgs: projectConfJson[elem].build?.compileArgs,
-        ninjaArgs: projectConfJson[elem].build?.ninjaArgs,
-        buildDirectoryPath: projectConfJson[elem].build?.buildDirectoryPath,
-        sdkconfigDefaults: projectConfJson[elem].build?.sdkconfigDefaults,
-        sdkconfigFilePath: projectConfJson[elem].build?.sdkconfigFilePath
-      },
-      env: projectConfJson[elem].env,
-      flashBaudRate: projectConfJson[elem].flashBaudRate,
-      monitorBaudRate: projectConfJson[elem].monitorBaudRate,
-      openOCD: {
-        debugLevel: projectConfJson[elem].openOCD?.debugLevel,
-        configs: projectConfJson[elem].openOCD?.configs,
-        args: projectConfJson[elem].openOCD?.args,
-      },
-      tasks: {
-        preBuild: projectConfJson[elem].tasks?.preBuild,
-        preFlash: projectConfJson[elem].tasks?.preFlash,
-        postBuild: projectConfJson[elem].tasks?.postBuild,
-        postFlash: projectConfJson[elem].tasks?.postFlash,
-      },
-    } as ProjectConfElement;
-  });
+  await Promise.all(
+    Object.keys(projectConfJson).map(async (elem) => {
+      const buildConfig = projectConfJson[elem].build;
+      const buildDirPath = buildConfig?.buildDirectoryPath
+        ? Uri.joinPath(workspaceFolder, buildConfig.buildDirectoryPath).fsPath
```

Comment:
Are all given paths relatives? What happens if the user defined a full path in `projectConfJson[elem].build.buildDirectoryPath`? 

I believe we should validate that the current value is a relative path. We could check that the current value with `await pathExists` before joining to the workspaceFolder.